project/test/test_sql_regression.py

```diff
@@ -0,0 +1,424 @@
+# -*- coding: iso-8859-1 -*-
+# pysqlite2/test/regression.py: pysqlite regression tests
+#
+# Copyright (C) 2006-2010 Gerhard H�ring <gh@ghaering.de>
+#
+# This file is part of pysqlite.
+#
+# This software is provided 'as-is', without any express or implied
+# warranty.  In no event will the authors be held liable for any damages
+# arising from the use of this software.
+#
+# Permission is granted to anyone to use this software for any purpose,
+# including commercial applications, and to alter it and redistribute it
+# freely, subject to the following restrictions:
+#
+# 1. The origin of this software must not be misrepresented; you must not
+#    claim that you wrote the original software. If you use this software
+#    in a product, an acknowledgment in the product documentation would be
+#    appreciated but is not required.
+# 2. Altered source versions must be plainly marked as such, and must not be
+#    misrepresented as being the original software.
+# 3. This notice may not be removed or altered from any source distribution.
+#
+# This is a modification of the CPython SQLite testbench for JPype.
+# The original can be found in cpython/Lib/sqlite3/test
+#
+
+
+import sys
+import _jpype
+import jpype
+from jpype.types import *
+from jpype import JPackage, java
+import common
+import pytest
+
+import datetime
+import unittest
+import jpype.dbapi2 as dbapi2
+import weakref
+import functools
+
+
+def getConnection():
+    return "jdbc:sqlite::memory:"
+
+
+class RegressionTests(common.JPypeTestCase):
+
+    def setUp(self):
+        common.JPypeTestCase.setUp(self)
+        self.con = dbapi2.connect(getConnection())
+
+    def tearDown(self):
+        self.con.close()
+
+    def testPragmaUserVersion(self):
+        # This used to crash pysqlite because this pragma command returns NULL for the column name
```

Comment:
It is not clear to me, why we want to test against regressions in pysqlite.

Reply:
But if this unit contains lots of useful cases, it might be worth checking them. I guess you have deleted unrelated cases already.

Reply:
Thanks!